ts/components/core/selection/CheckBox.tsx

```diff
@@ -0,0 +1,64 @@
+import * as React from "react";
+import { useState } from "react";
+import { StyleSheet } from "react-native";
+import TouchableDefaultOpacity from "../../TouchableDefaultOpacity";
+import IconFont from "../../ui/IconFont";
+import { calculateSlop } from "../accessibility";
+import { IOColors } from "../variables/IOColors";
+
+type Props = {
+  // the value of the checkbox
+  checked?: boolean;
+  // dispatch the new value after the checkbox changes state
+  onValueChange?: (newValue: boolean) => void;
```

Comment:
Could we support the disabled state? (untouchable)

`disabled?: boolean;`

Reply:
Added in #a20f46d (only logic state that blocks the interaction atm, waiting for graphical details to add a custom disable style).